### PR TITLE
Add check for java version when doing chmod on java and javac

### PR DESCRIPTION
**Pull request by Mesbah-Alam**
Signed-off-by: Mesbah <Mesbah_Alam@ca.ibm.com>

### CODE REVIEW

**Comment by Mesbah-Alam** [comment]
This change pertains to adding a check for java version when we are setting file permission for java and javac executables. The paths to these executables are different depending on Java version.

**Comment by Mesbah-Alam** [comment]
Personal build for testing this change: https://ci.adoptopenjdk.net/job/TestBuild_Sandbox2/99/console

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -67,8 +67,10 @@ def test() {
 			timestamps{
 				echo 'Building tests...'
 				sh "chmod 755 ${JAVA_BIN}/java"
-				sh "chmod 755 ${JAVA_BIN}/../../bin/javac"
-				sh "chmod 755 ${JAVA_BIN}/../../bin/java"
+				if (JAVA_VERSION == 'SE80') {
+					sh "chmod 755 ${JAVA_BIN}/../../bin/javac"
+					sh "chmod 755 ${JAVA_BIN}/../../bin/java"
+				}
```
Comment (sophia-guo) [inline buildenv/jenkins/JenkinsfileBase]: what if it's not SE80?

What's the difference between 
```sh "chmod 755 ${JAVA_BIN}/java"
sh "chmod 755 ${JAVA_BIN}/../../bin/java
```

Comment (Mesbah-Alam) [reply buildenv/jenkins/JenkinsfileBase]: When we are on SE80, 

`${JAVA_BIN}/java`=`$WORKSPACE/openjdkbinary/j2sdk-image/jre/bin/java` 
`${JAVA_BIN}/../../bin/java`=`$WORKSPACE/openjdkbinary/j2sdk-image/bin/java` 

While running on the internal IBM Java 8 SR6, I discovered that there are some tests that use the `"$WORKSPACE/openjdkbinary/j2sdk-image/bin/java` and `"$WORKSPACE/openjdkbinary/j2sdk-image/bin/javac`. That's why chmod'ing those executables, when we are on SE80, is required. 

The problem was only seen in Java 8 SR6 IBM internal builds, so, I will remove the chmod step which is outside of the if condition.

**Comment by sophia-guo** [comment]
Can't see my former comment. Recomment here:
Did you mean if it's SE90 or later the check is not necessary?

What's the difference between:
```
sh "chmod 755 ${JAVA_BIN}/java"

sh "chmod 755 ${JAVA_BIN}/../../bin/java"
```

**Comment by Mesbah-Alam** [comment]
@sophia-guo - I don't know how your comment and my reply to it got removed! 

Replying again here: 

When Java version is SE80: 
```
${JAVA_BIN}/java = $WORKSPACE/openjdkbinary/j2sdk-image/jre/binjava
${JAVA_BIN}/../../bin/java = $WORKSPACE/openjdkbinary/j2sdk-image/bin/java 
${JAVA_BIN}/../../bin/javac = $WORKSPACE/openjdkbinary/j2sdk-image/bin/javac
```

While attempting to run the tests on IBM SDK 8 SR6, I noticed that some tests refer to the last two executables above, as well as the first executable. Which is what prompted me to add those chmod statements. Since this issue was only seen on IBM SDK 8 SR6, when Java version is not SE80, the same chmod's are not required. I have refreshed the changeset accordingly. 
Does this answer your question?

**Comment by sophia-guo** [comment]
For SE80
we have j2sdk-image/jre/bin/java and j2sdk-image/bin/java 
For SE90
We only have j2sdk-image/bin/java

If you need to chmod for  both j2sdk-image**/jre/bin/j**ava and j2sdk-image**/bin**/java for SE80 I don't understand why for SE90 chmod of j2sdk-image**/bin**/java is not required


As java8 jre/bin/java  refer to

**Comment by Mesbah-Alam** [comment]
There seems to be an issue with IBM Java 8 SR6 SDK where the `javac` and `java` executables under `jre/bin` seem to require manual chmod to be executed. The same issue is not seen in OpenJ9 Java 8 SDK. I will reproduce the issue an open a bug against IBM Java 8 SR6.

**Comment by sophia-guo** [comment]
Approved supposing Personal build( including using template jenkinsfile) passed.

**Comment by Mesbah-Alam** [comment]
Personal build(triggered from mimic upstream job)
For SE80 : https://ci.adoptopenjdk.net/job/TestBuild_Sandbox2/104/console (fails after passing the stage we are checking for).

For SE90: https://ci.adoptopenjdk.net/job/TestBuild_Sandbox2/103/consoleFull

**Pull request merged**